docs/source/about/changelog.rst

```diff
@@ -23,7 +23,15 @@ more info, see the :ref:`Contributor Guide <Creating a Changelog Entry>`.
 Unreleased
 ----------
 
-No changes.
+**Changed**
+
+- :pull:`919` - Reverts :pull:`841` as per the conclusion in :discussion:`916`. but
+  preserves the ability to declare attributes with snake_case.
+
+**Deprecated**
+
+- :pull:`919` - Declaration of keys via keywork arguments in standard elements. A script
+  has been added to automatically convert old usages where possible.
```

Comment:
Do we want to expand this to components as well, so we can ultimately fix our component type hints?

Reply:
I'm not quite sure how doing this would fix component type hints. Also components have a completely different interface to standard elements - they're effectively a normal function call with the exception of the reserved `key` parameter.

Reply:
I recall us discovering that we can directly preserve type hints the original function's `*args, *kwargs` if we aren't trying to add a new kwarg (`key=...`).

Reply:
That's true, though we'd need a wrapper object around keys in order to distinguish them from other positional arguments. We could implement that as shown in [this gist](https://gist.github.com/5938469562e51a8c4617a8f472c4c684). We should definitely do this in a separate PR though.

Reply:
This also opens up other questions. Should we allow keys for standard elements to be declared in a similar manner? That is, as an optional first positional argument instead of via the normal attributes dict?

Reply:
> as an optional first positional argument instead of via the normal attributes dict

I think you might have mistyped